whois-api/src/main/java/net/ripe/db/whois/api/apiKey/ApiPublicKeyLoader.java

```diff
@@ -0,0 +1,103 @@
+package net.ripe.db.whois.api.apiKey;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.jakarta.rs.json.JacksonJsonProvider;
+import com.fasterxml.jackson.module.jakarta.xmlbind.JakartaXmlBindAnnotationIntrospector;
+import com.google.common.collect.Lists;
+import com.nimbusds.jose.util.JSONObjectUtils;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import com.nimbusds.jose.jwk.RSAKey;
+import jakarta.ws.rs.core.MediaType;
+import org.apache.commons.lang.StringUtils;
+import org.glassfish.jersey.client.ClientProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Nullable;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class ApiPublicKeyLoader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ApiPublicKeyLoader.class);
+
+    private static final int CLIENT_CONNECT_TIMEOUT = 10_000;
+    private static final int CLIENT_READ_TIMEOUT = 60_000;
+
+    private final Client client;
+    private final String restUrl;
+
+    @Autowired
+    public ApiPublicKeyLoader(
+            @Value("${api.public.key.url:}")  final String restUrl) {
+        this.restUrl = restUrl;
+
+        final ObjectMapper objectMapper = JsonMapper.builder()
```

Comment:
Move this setup code out of the constructor, into a static block or a separate method

Reply:
For now I can make a seperate method but I copied pasted from other class but it should go in a common place and every other restclient should use the same setting. If we are refactoring better to make it in a util function

Reply:
I agree most of that code is boilerplate and common to all clients, but some is different, e.g. the interceptors and the timeouts. I agree a utility function with parameters to customise it would be better than duplicating it.